### DISCUSSION TITLE
Livewire 3 legacy_model_binding as true brings problems surrounding  SupportLegacyModels

### DISCUSSION THREAD

**Discussion by JTibblesSoftcon** [livewire/livewire#7482] [7. Bugs (archived)]
### Livewire version

3.

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

There is a bug regarding SupportLegacyModels that somehow prevents Models with relationships from being returned after any event other than the initial component mount().

Steps to reproduce:

Switch SupportLegacyModels to true  in config/livewire.php

Example project, a seminar list, where parents (the users with a user id) can register their children to a seminar.
eg:

```
1) User table, standard Laravel users table with id, name, email etc...

2) Child table (user_id is the logged in user the "parent" of the "child"):
id   ||   user_id ||       firstname    ||     surname
  1  ||       1      ||         child 1      ||        Smith 
  2  ||       1      ||         child 2      ||        Smith 
     
3) Seminar table:
id   ||  title
10  ||  example seminar

4) UserSeminarAttendance table:
id   ||  user_id || seminar_id || child_id
 1   ||      1      ||         10      ||        1 
```

Create associated models for User, Child, Seminar and UserSeminarAttendance

Provide a "belongsTo" relationships inside UserSeminarAttendance to incorporate the Child model, so that the user will be able to return child firstname and surname via UserSeminarAttendance 

 ```
public function child(){
        return $this->belongsTo(Child::class, 'child_id', 'id');
}
```

Within a blank page create a new Livewire component ("details") and pass in a seminar id.

```
@livewire('details', [
       'seminarid' => 10,
])
```

In the details php file, create the mount function to return all children signed up to this seminar. Also create an empty function which we use shortly...:

```
public $userBookings;

public function mount(int $seminarid){
       $this->userBookings =  UserSeminarAttendance::where("user_id", Auth::user()->id)->where("seminar_id", $seminarid)->get();
}

public function tester(){
    //do nothing 
}
```

In the blade page simply export a list of children attending the seminar (using the userBookings  parameter). And below it, 
add an event button that, when clicked will call the "test" function:

```
@foreach($userBookings as $u)
    {{ $u->id }} : {{ $u->child->firstname }}, {{ $u->child->surname }} <br>
@endforeach

<button wire:click="tester">Do something/nothing</button>

```

Run the project.
You will see the list of children initially displays correctly.

BUT now click the button.......
You should now see an error along the lines of **"Class name must be a valid object or a string"**, even though there has been no change to the data.

This worked in Livewire 2 for sure. 

### Additional information

Interestingly if you convert the userBookings from an object to and array using ->toArray(), and change the blade properties to use an array rather than object, the error is gone but the child's name is no longer displayed because the Array can't store relational data, so there is no "->child" relationship anymore.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@JTibblesSoftcon can you share a screenshot of the error?

**Reply by JTibblesSoftcon** [reply]
unfortunately not, I've removed the legacy model flag as my current project is time critical and I need to keep moving with it. 
But I'll try and come back to this when my project is completed. I've found a couple of other interesting issues along the way which I'll add too.

**Reply by JTibblesSoftcon** [reply]
I managed to replicate the issue on my project again. For the error to occur, 'legacy_model_binding' = true.
When I switch it back to false it actually seems to work ok now. But the original models were created using Laravel 9 and Livewire 2.
Hope the screenshots help. From my understanding the error occurs during "HandleRequests@handleUpdate"

  ![image](https://github.com/livewire/livewire/assets/105914803/1f2af451-2be6-4a51-92f8-4b7459ce9c77)

![image](https://github.com/livewire/livewire/assets/105914803/27542b2b-2344-4d28-984d-7b4e5912fb56)

![image](https://github.com/livewire/livewire/assets/105914803/71d2e962-3efb-4837-8169-fc71f7419857)

![image](https://github.com/livewire/livewire/assets/105914803/92a10775-8607-4cf8-94bf-f002bf3eb899)

![image](https://github.com/livewire/livewire/assets/105914803/9734bf78-aec0-4004-8beb-206acc58f5e7)

**Reply by joshhanley** [reply]
@JTibblesSoftcon hmm yeah seems strange. In the first screenshot, can you expand the vendor frames and see if you can find the exact frame that is throwing the error and take a screenshot of that?

**Reply by JTibblesSoftcon** [reply]
![image](https://github.com/livewire/livewire/assets/105914803/52db5bb5-e41c-4c77-ae3a-ab8e982da89b)

These are the top ones, after that it's just the routes, pipelines etc. 
The problem seems to be to do with hydrating during updatehandling

**Reply by joshhanley** [reply]
@JTibblesSoftcon yeah the issue will be in the top one, the EloquentCollectionSynth. Can you browse to that and do a dump()/ray() call and dump all of the parameters of the `loadCollection()` method and send a screenshot of that? For some reason the `loadCollection()` isn't receiving the correct information, hence throwing the error.

**Reply by JTibblesSoftcon** [reply]
Doesn't look like it's receiving anything :/:

```
array:4 [▼ // vendor\livewire\livewire\src\Features\SupportLegacyModels\EloquentCollectionSynth.php:148
  "keys" => []
  "class" => "Illuminate\Database\Eloquent\Collection"
  "modelClass" => null
  "s" => "elcl"
]
```

for reference, the only data that is collected from DB at mount() for this livewire component is this:

`$userbookings = (Auth::user()) ? UserSeminarAttendance::where("user_id", Auth::user()->id)->where("seminar_id", $this->seminarid)->get() : [];`

The "UserSeminarAttendance" model has a child_id column which can be a number (the id of the child in the Child table) or null. If null then the user themselves are attending. If a child Id exists then use the Child relationship to return the name. 
The Blade page prints out the list of user bookings:

```
@foreach($userBookings as $u)
      @if(  (int) $u->child_id ?? 0 == 0 )
              {{ __('Adult is attending') }}
     @else
              {{ __('Child is attending: Name: ') }} {{  $u->child->nachname }}, {{ $u->child->vorname }}                                        
    @endif
@endforeach
```

**Reply by joshhanley** [reply]
@JTibblesSoftcon hmm ok yeah that is strange the the modelClass is empty. That's definitely what will be causing the issue. Is there a chance that your query is returning an empty EloquentCollection?
